test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/requirement/OpenShiftCommandLineToolsRequirement.java

```diff
@@ -23,48 +23,130 @@
+		 * Is in exclusion with downloadNewestOC.
+		 * @return false is default value
+		 */
+		boolean ocExists() default false;
+		
+		/**
+		 * Path to existing oc, ie. one that is configured within minishift setup-cdk
+		 * @return path to existing oc
+		 */
+		String pathToOC() default StringUtils.EMPTY;
+		
+		/**
+		 * Sets if default oc version or newest will be downloaded
+		 * @return true or false
+		 */
+		boolean downloadNewestOC() default false;
+		
+		/**
+		 * If set to true, given oc will be set into 
+		 * Preferences -> JBoss Tools -> OpenShift 3 -> oc executable location field
+		 * @return true or false
+		 */
+		boolean setOCInPrefs() default false;
+		
+		/**
+		 * Will delete downloaded oc, is not used when ocExists=true
+		 * @return true or false, default is true
+		 */
+		boolean cleanup() default true;
     }
+	
+	/**
+	 * Returns string representation of path to proper oc
+	 * @return
+	 */
+	public String getPathToOC() {
+		return this.pathToOC;
+	}
 
 	@Override
 	public void fulfill() {
-		if (!OCBinaryFile.get().getFile().exists()) {
-			// symlink does not exist or exists but points to inexistent file
-			File downloadedOCBinary = downloadAndExtractOpenShiftClient();
-			createSymlink(downloadedOCBinary);
+		if (!binary.ocExists()) {
+			if (!OCBinaryFile.get().getFile().exists()) {
+				String url = getDownloadLink(binary.downloadNewestOC());
+				LOGGER.info("OC binary will be downloaded from " + url);
+				File downloadedOCBinary = downloadAndExtractOpenShiftClient(url);
+				// windows has problem with previously defined symlinks, we will used copied oc from extracted 
+				// createSymLink method was used
+				this.pathToOC = copyOCFromExtractedFolder(downloadedOCBinary);
+			} else {
+				LOGGER.info("Binary is already downloaded.");
+				this.pathToOC = OCBinaryFile.get().getFile().getAbsolutePath();
+			}
 		} else {
-			LOGGER.info("Binary is already downloaded.");
+			LOGGER.info("External OC binary will be used");
+			File oc = new File(binary.pathToOC());
+			if (oc.exists()) {
+				LOGGER.info("OC binary is at " + oc.getAbsolutePath());
+				this.pathToOC = oc.getAbsolutePath();
+			} else {
+				throw new OpenShiftToolsException("Given path to OC binary does not exist: " + binary.pathToOC());
+			}
+		}
+		if (binary.setOCInPrefs()) {
+			setOCToPreferences(getPathToOC());
+		}
+	}
+	
+	private String copyOCFromExtractedFolder(File downloadedOCBinary) {
+		try {
+			return Files.copy(Paths.get(downloadedOCBinary.getAbsolutePath()),
+					OCBinaryFile.get().getFile().toPath(),
+					StandardCopyOption.REPLACE_EXISTING).toAbsolutePath().toString();
+		} catch (IOException e) {
+			throw new OpenShiftToolsException(NLS.bind("Could not copy {0} to {1}:\n{2}", 
+					new Object[] { OCBinaryFile.get().getFile().getAbsolutePath(), downloadedOCBinary.getAbsolutePath(), e }));
 		}
 	}
 
+	@SuppressWarnings("unused")
+	@Deprecated
 	private void createSymlink(File downloadedOCBinary) {
```

Comment:
@odockal This could be deleted. Use of this method leads to an error on Windows machines: "A required privilege is not held by the client..."